plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -417,6 +441,66 @@ private static LongReadFunction sqlServerDateReadFunction()
         return (resultSet, index) -> LocalDate.parse(resultSet.getString(index), DATE_FORMATTER).toEpochDay();
     }
 
+    private static ColumnMapping timestampWithTimeZoneColumnMapping(int precision)
+    {
+        checkArgument(precision <= MAX_DATETIMEOFFSET_PRECISION, "Unsupported datetimeoffset precision %s", precision);
+        if (precision <= TimestampWithTimeZoneType.MAX_SHORT_PRECISION) {
+            return ColumnMapping.longMapping(
+                    createTimestampWithTimeZoneType(precision),
+                    shortTimestampWithTimeZoneReadFunction(),
+                    shortTimestampWithTimeZoneWriteFunction());
+        }
+        return ColumnMapping.objectMapping(
+                createTimestampWithTimeZoneType(precision),
+                longTimestampWithTimeZoneReadFunction(),
+                longTimestampWithTimeZoneWriteFunction());
+    }
+
+    private static LongReadFunction shortTimestampWithTimeZoneReadFunction()
+    {
+        return (resultSet, columnIndex) -> {
+            DateTimeOffset dateTimeOffset = resultSet.getObject(columnIndex, DateTimeOffset.class);
+            ZonedDateTime zonedDateTime = dateTimeOffset.getOffsetDateTime().toZonedDateTime();
```

Comment:
`DateTimeOffset.getTimestamp()` gives a java.sql.Timestamp with nanos precision. Do we want to use it?

Reply:
This method is for short precision (<= 3), so we don't need nanos part in my opinion?